src/lib/mongoBackend/mongoListSubscriptions.cpp

```diff
@@ -23,42 +23,171 @@
+  if (throttling > -1)
+  {
+    // FIXME P10: int -> str conversion needed
+    s->notification.throttling.set("FIXME");
+  }
+}
+
+
+/* ****************************************************************************
+*
+* setDuration -
+*/
+extern void setDuration(Subscription* s, const BSONObj& r)
+{
+  //long long expiration = r.hasField(CSUB_EXPIRATION)? r.getField(CSUB_EXPIRATION).numberLong() : 0;
+  // FIXME P10: int -> str conversion needed
+  s->duration.set("FIXME");
+}
+
+/* ****************************************************************************
+*
+* mongoListSubscriptions -
+*/
 OrionError mongoListSubscriptions
 (
-  std::vector<Subscription>& subs,
+  std::vector<Subscription>&           subs,
   std::map<std::string, std::string>&  uriParam,
-  const std::string&                   tenant,
-  const std::vector<std::string>&      servicePathV
+  const std::string&                   tenant
 )
 {
-  OrionError oe;
-  oe.code = SccOk;
-
-  Subscription s1,s2;
-  EntID ei;
-  ei.id ="ID1";
-  s1.id ="s1";
-  s1.subject.entities.push_back(ei);
-  s1.subject.condition.attributes.push_back("A \nsc sub 1");
-  s1.subject.condition.attributes.push_back("B \tsc sub 1");
-  s1.subject.condition.attributes.push_back("Z \fsc sub 2");
-  s1.notification.attributes.push_back("notification attributes  1");
-  s1.notification.attributes.push_back("notification attributes 2");
-
-  subs.push_back(s1);
-
-  ei.idPattern ="IDP2";
-  ei.id = "";
-  ei.type ="FENOTIPO";
-  s2.id = "s2";
-  s2.subject.entities.push_back(ei);
-  s2.subject.condition.expression.q ="Q";
-  s2.subject.condition.expression.geometry ="G";
-  s2.subject.condition.expression.coords ="C";
-
-  subs.push_back(s2);
-
-  return oe;
+
+  // FIXME P10: Pagination not yet implemented
+
+  bool           reqSemTaken = false;
+
+  reqSemTake(__FUNCTION__, "Mongo List Subscriptions", SemReadOp, &reqSemTaken);
+
+  LM_T(LmtMongo, ("Mongo List Subscriptions"));
+
+  /* ONTIMEINTERVAL subscription are not part of NGSIv2, so they are excluded.
+   * Note that expiration is not taken into account (in the future, a q= query
+   * could be added to the operation in order to filter results) */
+  std::auto_ptr<DBClientCursor>  cursor;
+  std::string                    err;
+  std::string                    conds = std::string(CSUB_CONDITIONS) + "." + CSUB_CONDITIONS_TYPE;
+  BSONObj                        q = BSON(conds << "ONCHANGE");
+
+  if (!query(getSubscribeContextCollectionName(tenant), q, &cursor, &err))
+  {
+    reqSemGive(__FUNCTION__, "Mongo List Subscriptions", reqSemTaken);
+    return OrionError(SccReceiverInternalError, err);
+  }
+
+  /* Process query result */
+  while (cursor->more())
+  {
+    BSONObj r = cursor->next();
+    LM_T(LmtMongo, ("retrieved document: '%s'", r.toString().c_str()));
+
+    Subscription s;
+    setSubscriptionId(&s, r);
+    setSubject(&s, r);
+    setNotification(&s, r);
+    setDuration(&s, r);
```

Comment:
At the end, they are quite "tied" to the mongoBacked model, so they will stay as they are now (however, if we changer our mind in the future, it is very easy to move them to the ngsiv2::Subscription class).

NTC